albumentations/core/utils.py

```diff
@@ -65,12 +65,21 @@ def ensure_data_valid(self, data: Dict[str, Any]) -> None:
     def ensure_transforms_valid(self, transforms: Sequence[object]) -> None:
         pass
 
+    @abstractmethod
+    def convert_to_internal_type(self, data):
+        raise NotImplementedError
+
+    @abstractmethod
+    def convert_to_original_type(self, data):
+        raise NotImplementedError
+
     def postprocess(self, data: Dict[str, Any]) -> Dict[str, Any]:
         rows, cols = get_shape(data["image"])
 
         for data_name in self.data_fields:
-            data[data_name] = self.filter(data[data_name], rows, cols)
-            data[data_name] = self.check_and_convert(data[data_name], rows, cols, direction="from")
+            _data = self.filter(data[data_name], rows, cols, data_name)
+            _data = self.check_and_convert(_data, rows, cols, direction="from")
+            data[data_name] = self.convert_to_original_type(_data)
```

Comment:
I agree with combining `check_and_convert` with `convert_with_original_type` is more easier.
About moving `filter` after `convert_to_original_type` is what I don't understand. Since keypoints and bboxes will be filtered by this method in between each transforms, `filter` still need to be implemented in batch operation, in such cases, it'll have to be called before bboxes being converted back to their original form.

Reply:
Do we need to take DualIAATransform's implementation into considerations? Since currently, `apply_to_bboxes` in `DualIAATransform' can use `convert_bboxes_from_albumentations` to convert bboxes in batch to pascal_voc format, then use one for loop to convert bboxes into imgaug's BoundingBox object. If we combine `combine_bboxes_from_albumentations` with `convert_to_original_type` into something mentioned above, then in DualIAATransform's will inevitably require two for loop to convert internaltype to imgaug's BoundingBox object, which is slower compare to the current one.

Reply:
@Dipet friendly ping.